forwards_paper.tex

```diff
@@ -816,80 +816,134 @@ \subsubsection*{Sequential simplification and prior history}
-The space complexity after simplification is therefore $O(N + N\log N)$, since
-we need $O(N)$ space to store the first marginal tree, and $O(1)$ space
-for each of the $O(N \log N)$ trees that follow. Thus, we need $O(N^2)$ space
-to store the complete history of the simulation, but only $O(N + N \log N)$
-space to store the history that is relevant to the extant population.
+Therefore,
+we would need $O(N^2)$ space to store the complete history of the simulation,
+but only $O(N \log N)$ % $O(N + N \log N)$
+to store the history that is relevant to the extant population.
+
 
 %% Progressive memory reduction as a function of T
-\citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
-by retaining only 8 generations of history.
-This begs the question: how fast does the information in the pedigree become irrelevant?
-Suppose instead that we have run a simulation of $2N$ individuals for $T$ generations,
-begun with no prior history.
-One way to answer this question is to ask:
-How many of the resulting $4NT$ edges are required after simplification?
-As above, the expected number of edges is bounded by $2N-2$ plus four times the mean marginal tree length.
-Roughly, the length of time for which a marginal tree has $k$ tips is
-$2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations,
-and so the time to go from $N$ to $n$ tips is $2N(1/(n-1) - 1/(N-1))$.
-This implies that after running for time $T$ we expect to
-have around $n(T)$ roots, where $n(T) \approx 1 + 2N/(T+2)$.
+What if $T$ is smaller:
+how many of the resulting $2NT$ edges are required after simplification?
+In other words, how fast does the information in the pedigree become irrelevant?
+% \citet{padhukasahasram2008exploring} obtained impressive speedups relative to contemporary simulators
+% by retaining only 8 generations of history.
+% This begs the question: how fast does the information in the pedigree become irrelevant?
+Now, we need to compute the expected total length of all branches in a coalescent tree
+up until time $T$ (or the common ancestor, whichever comes first).
+Again, coalescent theory tells us that
+the expected length of time for which a coalescent tree has $k$ lineages
+is $2N/(k(k-1)) = 2N(1/(k-1) - 1/k)$ generations.
+Since the tree has $k$ branches over this period,
+it is expected to contribute $2N/(k-1)$ to the total tree length.
+By summing over $n < k \le N$,
+the $N$ tips of a tree are expected to descend from only $n$ lineages
+around $2(N/n - 1)$ generations ago.
+Inverting this relationship between time and number of roots
+implies that a marginal tree cut $T$ units of time ago
+% since 2N(1/n - 1/N) = 2(N/n-1) = T implies
+%    N/n = (T+2)/2 and so n = 2N/(T+2)
+is expected to have around $r(T)$ roots, where $r(T) \approx 2N/(T+2)$.
 The total tree length over this time is
-$2 N \sum_{k=n(T)+1}^{N-1} 1/k$, which
-% is approximately
+$2N \sum_{k=N}^{r(T)+1} 1/(k-1) \approx 2 N \log(N/r(T))$.
```

Comment:
Two issues here:
1. Are the upper and lower bounds mixed up? r(T) can't be > N.
2. More pedantically, r(T) isn't an integer, so is this a floor or a ceiling?

Reply:
1. Well,  I wrote it this way because it seems more natural to sum from now back until time T-ago; but good point, this might be confusing.
2. Yes, but let's sweep this under the rug, since below we change `\sum_{k=1}^x 1/k` to `log(x)` and it'd be nice to not propagate the floor.

Reply:
WRT to (2), basically I was trying to reproduce your log approximation here with wolfram alpha and not getting anywhere. Looking at it again though, I'm assuming your using some well-know result about \sum_x^y 1/(k - 1)  \approx log(x / (y - 1)}?

If so, great, I'm happy not to think about it!

Reply:
Since
```
\sum_{k=1}^n 1/k \approx \log(n) + \gamma
```
then
```
\sum_{k=x}^y 1/k = \sum_{k=1}^x 1/k - \sum_{k=1}^{y-1} 1/k
  \approx \log(x) + \gamma - \log(y) - \gamma
  = \log(x/y)
```
I'll add a line of explanation (without the `\gamma`).

Reply:
added small note here

Reply:
Ah, that makes sense, very nice.